wa/framework/target/runtime_config.py

```diff
@@ -26,67 +38,136 @@ def core_names(self):
+                                     setter_params={'core': name})
+
+        for cpu_no in range(self.target.number_of_cpus):
+            param_name = 'num_cpu{}_cores'.format(cpu_no)
+            self._runtime_params[param_name] = \
+                    RuntimeParameter(param_name, kind=int,
+                                     constraint=lambda x: 0 < x <= 1,
+                                     description="""
+                                     Specify whether cpu{} should be hotplugged
+                                     """.format(cpu_no),
+                                     setter=self.set_num_cores,
+                                     setter_params={'core': cpu_no})
+
+        if self.target.has('bl'):
+            for cluster in ['big', 'little']:
+                param_name = 'num_{}_cores'.format(cluster)
+                self._runtime_params[param_name] = \
+                        RuntimeParameter(param_name, kind=int,
+                                         constraint=lambda x, cluster=cluster:
+                                                   0 < x <= len(resolve_cpus(cluster), self.target),
+                                         description="""
+                                         The number of cores on the {} cluster to be hotplugged
+                                         """.format(cluster),
+                                         setter=self.set_num_cores,
+                                         setter_params={'core': cluster})
+
+    def check_target(self):
+        if not self.target.has('hotplug'):
+            raise TargetError('Target does not appear to support hotplug')
 
-        # cpus = cpusFromPrefix(core, self.target)
-        # core = name.split('_')[0]
-        value = integer(value)
-        if core not in self.core_names:
-            raise ValueError(name)
-        max_cores = self.core_count(core)
-        if value > max_cores:
-            message = 'Cannot set number of {}\'s to {}; max is {}'
-            raise ValueError(message.format(core, value, max_cores))
-        self.num_cores[core] = value
-        if all(v == 0 for v in self.num_cores.values()):
-            raise ValueError('Cannot set number of all cores to 0')
-
-    def set(self):
-        for c, n in reversed(sorted(self.num_cores.iteritems(),
-                                    key=lambda x: x[1])):
+    def validate_parameters(self):
+        if self.num_cores.values():
+            if all(v == 0 for v in self.num_cores.values()):
+                raise ValueError('Cannot set number of all cores to 0')
+
+    def commit(self):
+        '''Online all CPUs required before then off-lining'''
+        for c, n in self.num_cores.iteritems():
             self.set_num_online_cpus(c, n)
+        for c, n in self.num_cores.iteritems():
+            self.set_num_offline_cpus(c, n)
 
     def clear(self):
         self.num_cores = defaultdict(dict)
 
     def set_num_online_cpus(self, core, number):
```

Comment:
should be rnamed to "online_cpus()", as this doesn't set the number of online cpus, but onlines specific cpus.